Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp

```diff
@@ -245,9 +250,37 @@ static SkSamplingOptions toSkSamplingOptions(InterpolationQuality quality)
     return SkSamplingOptions(SkFilterMode::kLinear, SkMipmapMode::kNearest);
 }
 
-void GraphicsContextSkia::drawNativeImageInternal(NativeImage& nativeImage, const FloatRect& destRect, const FloatRect& srcRect, ImagePaintingOptions options)
+static sk_sp<SkImage> skiaImageFromNativeImage(GraphicsContextSkia& context, NativeImage& nativeImage)
```

Comment:
Maybe this could be a method of NativeImage? NativeImage::crossThreadPlatformImage or something like that?

Reply:
Maybe not NativeImage, but NativeImage -- I'll come up with a symmetric API for ImageBuffer/NativeImage.